src/main/scala/scorex/api/http/TransactionsApiRoute.scala

```diff
@@ -53,7 +54,12 @@ case class TransactionsApiRoute(
             case Some(txTypeStr) =>
               Exception.allCatch.opt(TransactionType(txTypeStr.toInt)) match {
                 case Some(txType: TransactionType.Value) =>
-                  complete(Json.obj("count" -> state.txTypeAccTxLengths(txType, a)))
+                  if(stateSettings.txTypeAccountTxIds){
+                    complete(Json.obj("count" -> state.txTypeAccTxLengths(txType, a)))
+                  }
+                  else {
+                    complete(unsupportedStateError("tx-type-account-tx-ids"))
+                  }
```

Comment:
this should be implemented in both `transactionCount` and `transactionList`?

is it possible to pick out these two APIs out and use a new file to implement the functions? and then we may add the `if` check in Application.scala?

Reply:
The api still works if tx type not specified